llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -12941,12 +12941,16 @@ ScalarEvolution::howManyLessThans(const SCEV *LHS, const SCEV *RHS,
       return RHS;
   }
 
-  // When the RHS is not invariant, we do not know the end bound of the loop and
-  // cannot calculate the ExactBECount needed by ExitLimit. However, we can
-  // calculate the MaxBECount, given the start, stride and max value for the end
-  // bound of the loop (RHS), and the fact that IV does not overflow (which is
-  // checked above).
   if (!isLoopInvariant(RHS, L)) {
+    // If RHS is an add recurrence, try again with lhs=lhs-rhs and rhs=0
```

Comment:
You can't subtract a value from the LHS and RHS of the comparison unless you can prove the subtraction doesn't wrap.

Reply:
@efriedma-quic Oh!
So suppose, we can prove that LHS Start < RHS Start and we know stride of lhs is positive and stride of RHS is negative then it should do the trick right?